python/lib/core/dmod/core/common/helper_functions.py

```diff
@@ -27,31 +32,31 @@ def get_mro_names(value) -> list[str]:
 
 
 def is_integer(value) -> bool:
-    if type(value) != type:
-        value = value.__class__
-
-    mro = get_mro_names(value)
+    if numpy and isinstance(value, numpy.integer):
+        return True
 
-    module_name = value.__module__ if hasattr(value, "__module__") else ''
+    loaded_integer_types = [int] + get_subclasses(int)
 
-    if module_name.startswith("numpy"):
-        return "integer" in mro
+    if numpy:
```

Comment:
I might be wrong, but I dont think this is necessary. `isinstance` is inclusive of subtypes.

Reply:
It is, but most of the numpy types don't inherit from the vanilla types. The numpy integer mro is: `(<class 'numpy.int32'>, <class 'numpy.signedinteger'>, <class 'numpy.integer'>, <class 'numpy.number'>, <class 'numpy.generic'>, <class 'object'>)`, which doesn't include `int`, so `numpy.int32(9).__class__ in [int] + int.__subclasses__()` is `False`.